lib/rubygems/commands/push_command.rb

```diff
@@ -52,23 +52,14 @@ def execute
     gem_name = get_one_gem_name
     default_gem_server, push_host = get_hosts_for(gem_name)
 
```

Comment:
This block was a branch and variable heavy way of saying use the user-provided host if available, or fall-back to the default host if none of the other options match. IMO the resultant single-if condition is clearer in intentions

Reply:
Yup. I understand the same thing. Can we extract this specific part of the PR to a separate commit and include your commit in the commit message?

Reply:
Yup, I understand the same thing. Can you extract the refactoring of this block to a separate commit and include your explanation in the commit description?

Reply:
Not sure wtf I'm doing but I seem to be deleting my own comments unintentionally. This is the third time I write this :rofl:.

So, I was saying that I agree with this and that it would be good if we could extract the refactoring of this block to a separate commit, and use your comment as the commit message.